### ISSUE TITLE
Warn on extra arguments with __aenter__

### ISSUE THREAD

**Issue by alexrudd2** [python-trio/flake8-async#421]
Today I realized that I had been unintentionally passing `*args` to an `__aenter_()`, which has no effect.  Docs here: https://docs.python.org/3/reference/datamodel.html#object.__aenter__

Perhaps a rule could have caught this?  I'm not aware of any valid use cases, and type checkers don't catch it either.

```python
async def __aenter__(self, *args) -> typing.self:   # warning: __aenter__() does not accept additional args
    return self

async def __aenter__(self, **kwargs) -> typing.self:   # warning: __aenter__() does not accept additional args
    return self

async def __aenter__(self) -> typing.self:  # OK
    return self
```

thank you for your consideration

**Comment by jakkdl** [comment]
I don't think flake8-async is the right place for this check, as it's much broader than just `__aenter__`. You can add on `*args` to the signature of any dunder function, be that `__enter__`, `__str__`, etc with the same behavior.
I'm not quite sure where it would be a good fit though. You could go ask the folks over in ruff if they have an opinion.

**Comment by jakkdl** [comment]
flake8-unused-arguments indeed seems perfect. I could see why ruff have opted into a different default for `*args` and `**kwargs` - but I think there's pretty strong case for them to then add a different rule that specifically warns about it, or have special behavior for dunders.
Upstreaming in general is definitely a good first approach, but given that upstream does handle it the next step would be opening a ruff issue. :)
